sdk/eventhubs/azure_messaging_eventhubs/src/common/recoverable_connection.rs

```diff
@@ -0,0 +1,1055 @@
+        unimplemented!("AmqpReceiverClient does not support detach operation");
+    }
+
+    async fn set_credit_mode(
+        &self,
+        _mode: azure_core_amqp::ReceiverCreditMode,
+    ) -> azure_core::Result<()> {
+        unimplemented!("AmqpReceiverClient does not support set_credit_mode operation");
+    }
+
+    async fn credit_mode(&self) -> azure_core::Result<azure_core_amqp::ReceiverCreditMode> {
+        unimplemented!("AmqpReceiverClient does not support credit_mode operation");
+    }
+
+    async fn receive_delivery(&self) -> azure_core::Result<azure_core_amqp::AmqpDelivery> {
+        let delivery = retry_azure_operation(
+            || async move {
+                let receiver = self
+                    .recoverable_connection
+                    .ensure_receiver(
+                        &self.source_url,
+                        &self.message_source,
+                        &self.receiver_options,
+                    )
+                    .await?;
+                if let Some(delivery_timeout) = self.timeout {
+                    select! {
+                        delivery = receiver.receive_delivery().fuse() => Ok(delivery),
+                        _ = azure_core::sleep::sleep(delivery_timeout).fuse() => {
+                             Err(azure_core::Error::new(
+                                AzureErrorKind::Io,
+                                Box::new(std::io::Error::from(std::io::ErrorKind::TimedOut))))
+                        },
+                    }?
+                } else {
+                    receiver.receive_delivery().await
+                }
+            },
+            &self.recoverable_connection.retry_options,
+            Some(RecoverableConnection::should_retry_receive_operation),
+        )
+        .await?;
+        Ok(delivery)
+    }
+
+    async fn accept_delivery(
+        &self,
+        _delivery: &azure_core_amqp::AmqpDelivery,
+    ) -> azure_core::Result<()> {
+        unimplemented!("AmqpReceiverClient does not support accept_delivery operation");
+    }
+
+    async fn reject_delivery(
+        &self,
+        _delivery: &azure_core_amqp::AmqpDelivery,
+    ) -> azure_core::Result<()> {
+        unimplemented!("AmqpReceiverClient does not support reject_delivery operation");
+    }
+
+    async fn release_delivery(
+        &self,
+        _delivery: &azure_core_amqp::AmqpDelivery,
+    ) -> azure_core::Result<()> {
+        unimplemented!("AmqpReceiverClient does not support release_delivery operation");
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{consumer, ErrorKind, EventHubsError};
+    use azure_core::credentials::TokenRequestOptions;
+    use azure_core::{credentials::AccessToken, http::Url, Result};
+    use azure_core_amqp::AmqpError;
+    use std::sync::{Arc, Mutex as SyncMutex};
+    use time::{Duration, OffsetDateTime};
+
+    // Helper struct to mock token credential
+    #[derive(Debug)]
+    struct MockTokenCredential {
```

Comment:
We have one of these in `azure_core_test` that might be of use and avoid duplication. Maybe. Can't remember off hand if it would do exactly what you want, but worth checking out.